### ISSUE TITLE
problem in usig gradtts decoder in unit speech

### ISSUE THREAD

**Issue by zeynabyousefi** [gmltmd789/UnitSpeech#9]
hi, 
thanks for valuabel repo
I test and inference from unit speech. but when I want to replace the decoder from the  main  Gradtts repo,the below error is ocuured!
```
  Missing key(s) in state_dict: "text_uncon", "spk_uncon", "estimator.downs.2.3.conv.weight", "estimator.downs.2.3.conv.bias", "estimator.downs.3.0.mlp.1.weight", "estimator.downs.3.0.mlp.1.bias", "estimator.downs.3.0.block1.block.0.weight", "estimator.downs.3.0.block1.block.0.bias", "estimator.downs.3.0.block1.block.1.weight", "estimator.downs.3.0.block1.block.1.bias", "estimator.downs.3.0.block2.block.0.weight", "estimator.downs.3.0.block2.block.0.bias", "estimator.downs.3.0.block2.block.1.weight", "estimator.downs.3.0.block2.block.1.bias",....
  Unexpected key(s) in state_dict: "estimator.spk_mlp.0.weight", "estimator.spk_mlp.0.bias", "estimator.spk_mlp.2.weight", "estimator.spk_mlp.2.bias". 
        size mismatch for estimator.mlp.0.weight: copying a param with shape torch.Size([256, 64]) from checkpoint, the shape in current model is torch.Size([512, 128]).
        size mismatch for estimator.mlp.0.bias: copying a param with shape torch.Size([256]) from checkpoint, the shape in current model is torch.Size([512]).
        size mismatch for estimator.mlp.2.weight: copying a param with shape torch.Size([64, 256]) from checkpoint, the shape in current model is torch.Size([128, 512]).
        size mismatch for estimator.mlp.2.bias: copying a param with shape torch.Size([64]) from checkpoint, the shape in current model is torch.Size([128]).
        size mismatch for estimator.downs.0.0.mlp.1.weight: copying a param with shape torch.Size([64, 64]) from checkpoint, the shape in current model is torch.Size([128, 384]).
        size mismatch for estimator.downs.0.0.mlp.1.bias: copying a param with shape torch.Size([64]) from checkpoint, the shape in current model is torch.Size([128]).
        size mismatch for estimator.downs.0.0.block1.block.0.weight: copying a param with shape torch.Size([64, 3, 3, 3]) from checkpoint, the shape in current model is torch.Size([128, 2, 3, 3]).
        size mismatch for estimator.downs.0.0.block1.block.0.bias: copying a param with shape torch.Size([64]) from checkpoint, the shape in current model is torch.Size([128]).
        size mismatch for estimator.downs.0.0.block1.block.1.weight: copying a param with shape torch.Size([64]) from checkpoint, the shape in current model is torch.Size([128]).
        size mismatch for estimator.downs.0.0.block1.block.1.bias: copying a param with shape torch.Size([64]) from checkpoint, the shape in current model is torch.Size([128]).,.....

```
how to manuplate the decoder of the gradtts to deploy it in unit speech?
bestregard
Zeynab Yousefi
@gmltmd789

**Comment by gmltmd789** [comment]
Hello,

First of all, while our model adopts the structure of GradTTS, there are several differences that prevent us from using the pre-trained GradTTS decoder directly.

1. We use a speaker embedding with d=256 to feed speaker conditions. This results in dimension mismatches between GradTTS and our model, which uses 256 dimensions.
2. We apply classifier-free guidance for both the speaker and text conditions. This is a conditioning enhancement technique for diffusion models that requires unconditional embeddings, leading to the introduction of new parameters called text_uncon and spk_uncon.
3. Our model is larger than GradTTS. Therefore, it is not possible to directly use the same decoder. For a decoder of the same size as GradTTS, you would need to set the decoder's dim and dim_mult to 64 and [1, 2, 4], respectively.
4. Additionally, our model requires a decoder trained on a variety of speakers, and there are other differences such as normalizing the data to a range of [-1, 1].

As such, we have made our English-trained decoder publicly available. If you're interested, we recommend referring to our model.

**Comment by zeynabyousefi** [comment]
Hello,

It seems that the decoder you used has a different architecture. I need to use this model for another language, but to do so, I have to train the decoder from scratch. The issue is that in GradTTS, there is no such thing as `dim_mult`, and I’m unsure how to proceed with training the decoder for a new language. Could you please guide me on how to train the decoder properly for using **Unit Speech** with another language?

Thank you.

**Comment by zeynabyousefi** [comment]
@gmltmd789

**Comment by gmltmd789** [comment]
I'll update the training code shortly :)
